app/policies/founder_policy.rb

```diff
@@ -3,7 +3,7 @@ def report?
     return false if user.blank?
 
     # School admins can view student profiles.
-    return true if user.school_admin.present?
+    return true if user.school_admin.present? && user.faculty.courses.exists?
```

Comment:
@a-c-sreedhar-reddy: I think we don't to check the presence of a `school_admin` at all and just have the faculty authorization alone which is already covered in the same function.